charger/easee.go

```diff
@@ -379,22 +387,94 @@ func (c *Easee) Enable(enable bool) error {
 		}
 
 		uri := fmt.Sprintf("%s/chargers/%s/settings", easee.API, c.charger)
-		resp, err := c.Post(uri, request.JSONContent, request.MarshalJSON(data))
-		if err != nil {
+		if err := c.postJSONAndWait(uri, data); err != nil {
 			return err
 		}
-		resp.Body.Close()
 	}
 
 	// resume/stop charger
 	action := easee.ChargePause
 	if enable {
 		action = easee.ChargeResume
 	}
+
 	uri := fmt.Sprintf("%s/chargers/%s/commands/%s", easee.API, c.charger, action)
-	_, err := c.Post(uri, request.JSONContent, nil)
+	if err := c.postJSONAndWait(uri, nil); err != nil {
+		return err
+	}
 
-	return err
+	if enable {
+		// reset currents after enable, as easee automatically resets to maxA
+		return c.MaxCurrent(int64(c.current))
+	}
+
+	return nil
+}
+
+// posts JSON to the Easee API endpoint and waits for the async response
+func (c *Easee) postJSONAndWait(uri string, data any) error {
+	resp, err := c.Post(uri, request.JSONContent, request.MarshalJSON(data))
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == 200 { //sync call
+		return nil
+	}
+
+	if resp.StatusCode == 202 { //async call, wait for response
+		var cmd easee.RestCommandResponse
+
+		if strings.Contains(uri, "/commands/") { //command endpoint
+			if err := decodeJSON(resp, &cmd); err != nil {
+				return err
+			}
+		} else { //settings endpoint
+			var cmdArr []easee.RestCommandResponse
+			if err := decodeJSON(resp, &cmdArr); err != nil {
+				return err
+			}
+
+			if len(cmdArr) != 0 {
+				cmd = cmdArr[0]
+			}
+		}
+
+		if cmd.Ticks == 0 { //Easee API thinks this was a noop
+			return nil
+		}
+		return c.waitForTickResponse(cmd.Ticks)
+	}
+
+	// all other response codes lead to an error
+	return fmt.Errorf("invalid status: %d", resp.StatusCode)
+}
+
+// decodeJSON reads HTTP response and decodes JSON body if error is nil
+func decodeJSON(resp *http.Response, res interface{}) error {
```

Comment:
Brauchen wir das einschl. des `ResponseError` noch? Es kann ja nur noch Status 202 sein, ergo kein weiterer Error mehr auftauchen? sah beim letzten Anlauf schon komisch aus, jetzt weiss ich warum :)

Reply:
Jetzt wo du es erwähnst... stimmt. Das kann raus. Dann können wir die Funktion aber auch inlinen - für den Einzeiler lohnt es nicht mehr.

Bonus Frage: Macht das decoding im Fehlerfall in L457 überhaupt Sinn...? Ich habe das schamlos aus helper.go geklaut, da wird das auch so gemacht, siehe https://github.com/evcc-io/evcc/blob/aaa48022e8fcacfc88666825095bacd6c93a7c24/util/request/helper.go#L58 - aber wenn ein Fehler auftritt, warum dann überhaupt noch decoden?